web/sidebar/SidebarRoomHeader.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+
+export class SidebarRoomHeader extends Component {
+  constructor() {
+    super();
+    this.toggleRoomInput = this.toggleRoomInput.bind(this);
+    this.mouseOver = this.mouseOver.bind(this);
+    this.mouseOut = this.mouseOut.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handlePressedEnter = this.handlePressedEnter.bind(this);
+
+    this.state = {
+      value: '',
+      hover: false,
+      isInputVisible: false
+    };
+  }
+
+  getRoomInput() {
+    const groupStyles = { margin: '5px 0' };
+
+    return (
+      <div className="input-group" style={groupStyles}>
+        <div className="input-group-addon">#</div>
+        <input type="text" className="form-control input-sm" placeholder="subreddit"
+               value={ this.state.value }
+               onChange={ this.handleChange }
+               onKeyPress={ this.handlePressedEnter }
+        />
+      </div>
+    );
+  }
+
+  toggleRoomInput() {
+    const isVisible = !this.state.isInputVisible;
+    this.setState({ isInputVisible: isVisible });
+    if(!isVisible) this.setState({ value: "" });
+
+  }
+
+  mouseOver() {
+    this.setState({ hover: true });
+  }
+
+  mouseOut() {
+    this.setState({ hover: false });
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handlePressedEnter(event) {
```

Comment:
You should trigger `chat-actions.handleChangeRoom` here instead of doing it manually to make sure we're changing rooms all in one play. Also `handleChangeRoom` does some other things as well like update the `lastMessageTime` for the room you change to.